File: Day_28/Q111.cpp
#include <bits/stdc++.h>
using namespace std;

//Structure For Ticket Details
struct Ticket{
    int ticketId;
    string name;
    string movie;
};

//Book New Ticket
void bookTicket(vector<Ticket> &t){

    Ticket x;

    cout << "Enter Ticket ID : ";
    cin >> x.ticketId;

    cin.ignore();

    cout << "Enter Customer Name : ";
    getline(cin,x.name);

    cout << "Enter Movie Name : ";
    getline(cin,x.movie);

    t.push_back(x);

    cout << "Ticket Booked Successfully\n";
}

//Display All Tickets
void displayTickets(vector<Ticket> t){

    if(t.size() == 0){
        cout << "No Ticket Available\n";
        return;
    }

    for(int i = 0;i < t.size();i++){

        cout << "\nTicket ID : " << t[i].ticketId << endl;
        cout << "Customer Name : " << t[i].name << endl;
        cout << "Movie Name : " << t[i].movie << endl;
    }
}

//Search Ticket By Ticket ID
void searchTicket(vector<Ticket> t){

    int id;
    bool found = false;

    cout << "Enter Ticket ID : ";
    cin >> id;

    for(int i = 0;i < t.size();i++){

        if(t[i].ticketId == id){

            cout << "\nTicket Found\n";
            cout << "Customer Name : " << t[i].name << endl;
            cout << "Movie Name : " << t[i].movie << endl;

            found = true;
        }
    }

    if(found == false) cout << "Ticket Not Found\n";
}

int main(){

    vector<Ticket> t;
    int choice;

    do{

        //Menu Driven Program
        cout << "\n1. Book Ticket";
        cout << "\n2. Display Tickets";
        cout << "\n3. Search Ticket";
        cout << "\n4. Exit";
        cout << "\nEnter Choice : ";
        cin >> choice;

        if(choice == 1) bookTicket(t);

        else if(choice == 2) displayTickets(t);

        else if(choice == 3) searchTicket(t);

    }while(choice != 4);

    return 0;
}

File: Day_28/Q111.java
import java.util.*;

//Class For Ticket Details
class Ticket{
    int ticketId;
    String name;
    String movie;
}

class Q111{

    static Scanner sc = new Scanner(System.in);

    //Book New Ticket
    static void bookTicket(ArrayList<Ticket> t){

        Ticket x = new Ticket();

        System.out.print("Enter Ticket ID : ");
        x.ticketId = sc.nextInt();

        sc.nextLine();

        System.out.print("Enter Customer Name : ");
        x.name = sc.nextLine();

        System.out.print("Enter Movie Name : ");
        x.movie = sc.nextLine();

        t.add(x);

        System.out.println("Ticket Booked Successfully");
    }

    //Display All Tickets
    static void displayTickets(ArrayList<Ticket> t){

        if(t.size() == 0){
            System.out.println("No Ticket Available");
            return;
        }

        for(int i = 0;i < t.size();i++){

            System.out.println("\nTicket ID : " + t.get(i).ticketId);
            System.out.println("Customer Name : " + t.get(i).name);
            System.out.println("Movie Name : " + t.get(i).movie);
        }
    }

    //Search Ticket By Ticket ID
    static void searchTicket(ArrayList<Ticket> t){

        int id;
        boolean found = false;

        System.out.print("Enter Ticket ID : ");
        id = sc.nextInt();

        for(int i = 0;i < t.size();i++){

            if(t.get(i).ticketId == id){

                System.out.println("\nTicket Found");
                System.out.println("Customer Name : " + t.get(i).name);
                System.out.println("Movie Name : " + t.get(i).movie);

                found = true;
            }
        }

        if(found == false) System.out.println("Ticket Not Found");
    }

    public static void main(String args[]){

        ArrayList<Ticket> t = new ArrayList<>();

        int choice;

        do{

            //Menu Driven Program
            System.out.println("\n1. Book Ticket");
            System.out.println("2. Display Tickets");
            System.out.println("3. Search Ticket");
            System.out.println("4. Exit");

            System.out.print("Enter Choice : ");
            choice = sc.nextInt();

            if(choice == 1) bookTicket(t);

            else if(choice == 2) displayTickets(t);

            else if(choice == 3) searchTicket(t);

        }while(choice != 4);

        sc.close();
    }
}
